client/my-sites/stats/stats-plan-usage/index.tsx

```diff
@@ -100,16 +120,18 @@ const PlanUsage: React.FC< PlanUsageProps > = ( {
 				</div>
 			</div>
 			<div className="plan-usage-note">
-				<span>
-					{ overLimitMonthsText } { upgradeNote }
-				</span>
+				<span>{ overLimitMonthsText }</span>
+				{ upgradeNote }
 			</div>
 		</div>
 	);
 };
 
 const StatsPlanUsage: React.FC< StatsPlanUsageProps > = ( { siteId, isOdysseyStats } ) => {
 	const upgradeLink = getStatsPurchaseURL( siteId, isOdysseyStats );
+	const upgradeEvent = `${
```

Comment:
As another [similar tier upgrade action event](https://github.com/Automattic/wp-calypso/blob/8bf44a682056cd95b2b19dc07e34f5f6ece3f368/client/my-sites/stats/stats-notices/tier-upgrade-notice.tsx#L50-L52) is triggered in the `TierUpgradeNotice`, can the event tracking of upgrade actions be consolidated to the purchase page viewing by checking the `from` [param on the query string](https://github.com/Automattic/wp-calypso/blob/8bf44a682056cd95b2b19dc07e34f5f6ece3f368/client/my-sites/stats/stats-notices/tier-upgrade-notice.tsx#L11)?

Reply:
Yes, I can consolidate it, but I think it will be messy. 
It will have to happen in <PageViewTracker /> to avoid additional events triggered on re-rendering.
We will change the purpose of the component, and I think that, what I proposed here, is closer to the source of the event.
If we decide to extend PageViewTracker I would have to apply the additional tracking only in two cases (upgrade notice and usage bar) and exclude other `from` values which would require an additional check.

I'm not sure if that's a better solution and the code proposed here is easier to notice.

Reply:
> what I proposed here, is closer to the source of the event.

IMHO, clicking on the CTA eventually enters the purchase page, which would be more meaningful to the purchase flow. E.g., How many people view the page, and what is the conversion rate to the checkout page? We can choose to do the recording in one place with checks for the query `from`.

Of course, being close to the source of the event is more reliable for accurate recording. So, I'm just going to defer to your decision. 🙂

Reply:
I like your point about tracking the checkout conversion rate, but checkout redirects to the main traffic page, not back to the purchase page. Did you have something else in mind?

On the other hand, I also see your point that it's more important to track events that actually viewed the purchase page rather than just clicking a link.

I'll give it a try with `useEffect` and keep it out of the page views component.